ui/plugin-system/src/runtime/plugin-registry.ts

```diff
@@ -82,3 +83,18 @@ export function useListPluginMetadata(pluginType: PluginType, options?: UseListP
   const { listPluginMetadata } = usePluginRegistry();
   return useQuery(['listPluginMetadata', pluginType], () => listPluginMetadata(pluginType), options);
 }
+
+export async function usePluginBuiltinVariableDefinitions() {
+  const { getPlugin, listPluginMetadata } = usePluginRegistry();
+
+  const datasources = await listPluginMetadata('Datasource');
+  const datasourceKinds = new Set(datasources.map((datasource) => datasource.kind));
+  let result: BuiltinVariableDefinition[] = [];
+  for (const kind of datasourceKinds) {
+    const plugin = await getPlugin('Datasource', kind);
+    if (plugin.getBuiltinVariableDefinitions) {
+      result = result.concat(plugin.getBuiltinVariableDefinitions());
```

Comment:
Maybe we'll want to keep somewhere the information of from which datasource plugin the builtin variable was coming from. 

To do that you will certainly have to transform ``result`` to a list of list instead of just a list

Reply:
I plan to add it in the BuiltinVar spec